core/tls.go

```diff
@@ -102,5 +110,24 @@ func LoadTrustStore(trustStoreFile string) (*TrustStore, error) {
 		}
 	}
 
+	if err = validate(trustStore); err != nil {
+		return nil, err
+	}
+
 	return trustStore, nil
 }
+
+// validate returns an error if one of the certificates is invalid or does not form a chain to some root
+func validate(store *TrustStore) error {
+	opts := x509.VerifyOptions{
+		Intermediates: NewCertPool(store.IntermediateCAs),
+		Roots:         NewCertPool(store.RootCAs),
+		CurrentTime:   nowFunc(), // leaving this empty defaults to the current time. Cannot not check time.
+	}
+	for _, cert := range store.Certificates() {
+		if _, err := cert.Verify(opts); err != nil {
+			return err
```

Comment:
does it fail when one of the trust store certificate's is expired? I don't think we want that. Since that might cause failures just because the next day arrived

Reply:
Yes it fails when a cert has expired. However, this check is only performed at startup, so it wil not cause random crashes on a node that is already running. In that case it will start producing warnings every hour from the CRL validator. Failing to start with an insecure setup sounds like a save default?

Reply:
It's not insecure; certificates will be rejected because the CA certs are expired. I think we should just log a warning. Can't we skip the expiry check?